### ISSUE TITLE
Make unacceptable geopoint accuracy cutoff configurable

### ISSUE THREAD

**Issue by lognaturel** [getodk/collect#4961]
```
 category = if accuracy > unnacceptableAccuracyThreshold
    unacceptable
else if accuracy > accuracyThreshold + 5
    poor
else
    improving
```
<details>
  <summary>Brainstorming</summary>
Currently there is a accuracyThreshold body attribute: “For <input> with type geopoint, geotrace, or geoshape this sets the auto-accept threshold in meters for geopoint captures”. I’m not entirely sure how/if this is actually supported for traces and shapes. 

We want to also make the unacceptable accuracy cutoff configurable. I think by analogy to `accuracyThreshold` it should be a body attribute in camel case. Additionally, I think we should put it in the http://www.opendatakit.org/xforms namespace. For now it should only apply to manual geopoint.  

I propose `odk:worstAcceptableAccuracy`

Note: I think these are appropriate as body attributes rather than bind attributes because they're more like user experience concerns rather than data concerns. In the case of this new attribute, it can be entirely ignored by a client that doesn't have the new dialog with no data difference.

I think implementation can happen even if we’re not totally sure about the name since it would be easy enough to change. What users will really interact with will be XLSForm which we can use a different name for. 

If the attribute is set, its value is used to determine when to go past the unacceptable accuracy step in the geopoint dialog. If it’s not set, 100m is used. If it is set but is less than accuracyThreshold or accuracyThreshold + 5, then the unacceptable accuracy step on the progress bar is always filled in and the text never is displayed. 
<details>

**Comment by seadowg** [comment]
@lognaturel Going to go with `unacceptableAccuracyThreshold` for the moment instead as it feels like it mirrors both the other param and the language used in the app. We can easily switch back if you don't agree!

**Comment by lognaturel** [comment]
> Going to go with unacceptableAccuracyThreshold for the moment 

Works for me! I’m always torn between matching existing naming and trying for something better. Relatedly, we’ve been using dash-separated words for attribute names recently. I believe that was to match W3C XForms conventions. Here since we already have a related attribute that uses lower camel case here I didn’t think too much about it and figured we should match. Just making that explicit. 

> If it is set but is less than accuracyThreshold or accuracyThreshold + 5, then the unacceptable accuracy step on the progress bar is always filled in and the text never is displayed.

Is this clear and reasonable to you? If so I want to use it in our response to forum feedback. I could see it feeling kind of backwards but I can’t think of another good way to handle that case. I also think we do want a way to skip the “unacceptable” message.

**Comment by seadowg** [comment]
> If it is set but is less than accuracyThreshold or accuracyThreshold + 5, then the unacceptable accuracy step on the progress bar is always filled in and the text never is displayed.

Was just moving on to this part and don't quite understand it. "the text never is displayed" - not sure what this means? I was thinking that if the value is less than the other thresholds then the accuracy just always shown as "unacceptable" (until it's auto saved).

**Comment by lognaturel** [comment]
> if the value is less than the other thresholds then the accuracy just always shown as "unacceptable" (until it's auto saved).

Yeah, ok, I guess that makes more sense. And if a user never wants to see "unacceptable" (what I want to make sure is supported) they just need to put an outrageously high value for the unacceptable threshold. Not sure why I wasn't seeing that as a reasonable approach.

I've updated the issue with what I think are the requirements we've agreed on. Please update if anything seems off.

**Comment by seadowg** [comment]
> Please update if anything seems off.

I switched the language round so "higher" accuracy is a greater number ("accuracy - 5" becomes "accuracy + 5" for instance). I find it pretty impossible to reason about otherwise, although I see how you got there.

**Comment by seadowg** [comment]
> If the attribute has a value that is less than or equal to accuracyThreshold + 5, the "unacceptable" message and color are shown until the accuracy reaches the accuracyThreshold and the point is auto captured

@lognaturel I'm not sure if I understand the reasoning here. There are three categories for accuracy above the threshold: improving (<= accuracyThreshold + 5), poor (> accuracyThreshold + 5) and unacceptable (> 100 or unacceptableAccuracyThreshold). Here you're saying that if unacceptableAccuracyThreshold <= accuracyThreshold + 5 then we never get to "improving", even if unnacceptableAccuracyThreshold > accuracyThreshold. I think this is potentially best written out in (pseudo) code, so there's precedence:

```
category = if accuracy > unnacceptableAccuracyThreshold
    unacceptable
else if accuracy > accuracyThreshold + 5
    poor
else
    improving
```

So basically if accuracy is greater than unnacceptableAccuracyThreshold we show it as unacceptable and otherwise we fallback to checking if we show it as poor or improving. Does that look right to you? I'm going to assume this was the intention and move forward.

**Comment by lognaturel** [comment]
> I'm not sure if I understand the reasoning here. 

Sigh, why write something simple when I can write something inscrutable! Yes, you did it the sanest way.
